### PR TITLE
Feature/validation and model specs

### PR DESCRIPTION
**Pull request by blessedjasonmwanza**
In this TDD, following [guidlines](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n+1.md)
- Add validators to all models
- Add Specs unit tests for all models
- Fetch Limited data to prevent N+1 problems
- Fix linter errors
- Add flash messages when failing to save info e.g in Create Article post

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+    post = Post.new(title: 'New test post', text: 'How are ', author: author, likes_counter: 0, comments_counter: 0)
+    post.save!
+
+    comment_creator = User.new(name: 'Jerry', photo: 'profile_pic.jpeg', bio: 'Bio template', posts_counter: 0)
+    comment_creator.save!
+
+    post.comments.create!(text: 'Hi World', author: comment_creator)
+    post.comments.create!(text: 'This is actually my second post', author: comment_creator)
+    expect(post.comments.length).to eql(2)
+  end
```
Comment (Emmyn5600) [inline spec/models/comment_spec.rb:15]: - All the tests are not running successfully, please fix it.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @blessedjasonmwanza 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
 - You didn't solve the N+1 problem when fetching all posts and their comments for a user. Please fix it.
- Check the comment under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (blessedjasonmwanza) [reply spec/models/comment_spec.rb:15]: Hi @Emmyn5600 Which branch are you running?

The code you shared in the snippet is not what I wrote. Check the ReadMe file as well on how to run test.

Comment (blessedjasonmwanza) [reply spec/models/comment_spec.rb:15]: Hi @Emmyn5600,

I have fixed the issues. 

Thanks a lot 👨🏽‍💻 />

**Review by dasileker — APPROVED** [review]
### **Hi @blessedjasonmwanza,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**